src/main.js

```diff
@@ -29,60 +30,71 @@ async function run() {
         if (data.TrxData.TestRun.ResultSummary._outcome === 'Failed') {
           conclusion = ignoreTestFailures ? 'neutral' : 'failure';
         }
-        await createStatusCheck(token, data, markupData, conclusion);
+        const checkId = await createStatusCheck(token, data, markupData, conclusion);
+        statusCheckIds.push(checkId);
       }
-
-      markupForComment.push(markupData);
+    }
+    if (shouldCreateStatusCheck && statusCheckIds.length > 0) {
+      core.info(`\nThe following status check ids were created: ${statusCheckIds.join(',')}`);
+      core.setOutput('status-check-ids', statusCheckIds.join(',')); // This is mainly for testing purposes
     }
 
-    // The README.md indicates only one per comment per run (so all trx files found)
-    if (markupForComment.length > 0 && shouldCreatePRComment) {
-      const commentCharacterLimit = 65535;
-      let markupComment = markupForComment.join('\n');
-      if (markupComment.length > commentCharacterLimit) {
-        core.info(
-          `Truncating markup data due to character limit exceeded for github api.  Markup data length: ${markupComment.length}/${commentCharacterLimit}`
-        );
-        markupComment = markupComment.substring(0, commentCharacterLimit - 100);
-        markupComment = 'Test outcome truncated due to character limit. See full report in output. \n' + markupComment;
-        core.setOutput('test-outcome-truncated', 'true');
-      } else {
-        core.setOutput('test-outcome-truncated', 'false');
+    // 4 - Create a PR comment if requested
+    if (markupForResults.length > 0 && shouldCreatePRComment) {
+      let markup = markupForResults.join('\n');
+      core.info(`\nCreating a PR comment with length ${markup.length}...`);
+
+      // GitHub API has a limit of 65535 characters for a comment so truncate the markup if we need to
+      const charLimit = 65535;
+      let truncated = false;
+      if (markup.length > charLimit) {
+        const message = `Truncating markup data due to character limit exceeded for GitHub API.  Markup data length: ${markup.length}/${charLimit}`;
+        core.info(message);
+
+        markup = markup.substring(0, charLimit - 100);
+        markup = 'Test results truncated due to character limit. See full report in output. \n' + markup;
+        truncated = true;
       }
+      core.setOutput('test-results-truncated', truncated);
 
-      // TODO:  implement steve's change for cypress
-      await createPrComment(token, markupComment, updateCommentIfOneExists, commentIdentifier);
+      const commentId = await createPrComment(token, markup, updateCommentIfOneExists, commentIdentifier);
+      core.setOutput('pr-comment-id', commentId); // This is mainly for testing purposes
     }
 
+    // 5 - Create a results file if requested
     if (shouldCreateResultsFile) {
+      // QUESTION:  if this is called multiple times in one job the file contents will be replaced for each instance.  Should we fix that?
```

Comment:
I imagine since no one has brought up this deficiency that no one is calling it multiple times in a single job, so it's probably not a huge priority to fix. Do we have a warning somewhere more visible though? That might be useful if not.

Reply:
Good suggestion, I've added it to the README.md.